code/modules/reagents/reagent_containers/pill.dm

```diff
@@ -31,34 +32,27 @@
 	return FALSE
 
 /obj/item/reagent_containers/food/pill/afterattack(obj/target, mob/user, proximity)
-	if(!proximity)
+	if(!proximity || !target.is_open_container())
+		return
+	else if(!target.reagents.total_volume)
+		to_chat(user, "<span class='warning'>[target] has no liquid to dissolve [src] in.</span>")
```

Comment:
I sometimes wonder if we should just get rid of this. It doesn't seem to server a particularly interesting balance function---I know what they're going to, in terms of realism, but being able to just crack a pill open and empty into an empty beaker doesn't seem like it's going to terribly break anything.

Reply:
You have a point. Sinks are aplenty across the station so it wouldn't be hard to add some water to the beaker to dissolve the pill in. At best this is a mild inconvenience.